clients/src/main/java/org/apache/kafka/clients/consumer/internals/KafkaConsumerMetrics.java

```diff
@@ -78,4 +79,11 @@ public void recordPollEnd(long pollEndMs) {
         double pollIdleRatio = pollTimeMs * 1.0 / (pollTimeMs + timeSinceLastPollMs);
         this.pollIdleSensor.record(pollIdleRatio);
     }
+
+    @Override
+    public void close() {
```

Comment:
I think we need a unit test that passes mock Metrics to C'tor and validates the number of addMetric and removeMetric match. Otherwise we may introduce a new memory leak.

Reply:
Agreed.  I added a test.